torch/csrc/distributed/c10d/reducer.cpp

```diff
@@ -2308,4 +2308,17 @@ void Reducer::check_finalized() {
   ensure_prior_reduction_finished();
 }
 
+void Reducer::update_process_group(
+    c10::intrusive_ptr<c10d::ProcessGroup> new_process_group) {
+  std::lock_guard<std::mutex> lock(mutex_);
+  process_group_ = std::move(new_process_group);
```

Comment:
should we validate that the new PG matches the old PG in important ways (such as size)?